Проверка страницы Product.html

StasOskol opened the pull request.

StasOskol committed:
Создание новой папки и нового файла Product.html

В Product.html находится код, что и на Catalog.html, так как они очень похожи

StasOskol committed:
Подготовка шаблона по страницу Product.html

Уборка и подготовка шаблона под страницу Product.html.
Оставлен был только один ряд карточек, были убраны фильтры и кнопки, всё остальное осталось таким же как и в Catalog.html, так как страницы похожи

StasOskol committed:
Предпоследний этап работы со страницей Product

1. Добавление изображения на фронт страницы
2. Добавление секции (карусель), эта секция имеет position:relative; - прикрепление к этому блоку
3. Ограничения блока по размерам
4. Добавление двух кнопок вращения карусели (position:absolite)
5. При адаптиве страницы - внимательнее с кнопками
6. Добавление блока (описание тела карусели) - position:absolute;, но нахождение блока внутри секции
7. Выравнивание блока по центру и ограничения внутри
8. Добавление содержимого описания front карусели
9.  Кнопка с содержимым и эффектами наведения
10. В описании присутсвуют фильтры

StasOskol committed:
Последний этап со страницей Product

Изменение картинок под содержимое страницы Product из макета

Окончание работы со страницей Product

StasOskol committed:
Merge pull request #17 from StasBeep/checkTeacher

Check teacher

Вёрстка страницы catalog завершена
В странице index.html (как и на странице Catalog.html) были добавлены эффекты наведения на карточки с последующим показом кнопки покупки (кнопка с корзиной и текстом))
Для фильтров применены теги details и summary

Есть вопрос:
При наведении на карточку товара появляется кнопка (кнопка с корзиной и текстом), но применить эффекты к блокам внутри кнопки не получается, подскажите как это сделать?

StasOskol committed:
Страница прошла типограф и валидатор (Product.html)

StasOskol committed:
Merge pull request #18 from StasBeep/WorkProduct

Work product end

Завершение работы со страницей Product

Страница прошла типограф и валидатор

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+<!DOCTYPE html>
```
имена файлов и папок маленькими буквами

StasOskol replied:
Вопрос: там есть кадировка "UTF-8" - оно с прописных символов, исправление может привести к плохим последствиям, это надо исправлять?

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+                        <!--Правая часть меню-->
+                        <div class="right-side">
+                            <!--Выползающее меню (смотреть позже)-->
+                            <button class="right-side-crawling-menu">
```
type="button" указывйте тип у кнопок здесь и далее по файлику

StasOskol replied:
type="button" нужно указывать после класса или до ? Или это не принципиально ?  И ещё вопросик: это правило "хорошего тона" так делать требует?

VanillaWulf replied:
без разницы, но обычно сначала идут класс, потом атрибуты

VanillaWulf replied:
по дефолту у кнопок submit, но по правилам хорошего тона указывают тип

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+                    </div>
+                </nav>
+                <!--Второй блок под меню-->
+                <div class="bgcolorp">
```
main>section

StasOskol replied:
Здесь этот блок исправил под section, но вот не понял зачем нужно его помещать в main, везде делал так, чтобы первый блок относился к header, зачем это изменять?

div на section заменил, а помещать этот блок в main не стал

VanillaWulf replied:
если блок дублируется на страницах всех, то его в хедер, если он есть на парочке страниц - то мейн

VanillaWulf replied:
мейн - это главный блок, оторванных секций от структурных элементов не должно быть

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+                    <div class="wrapper padding-title">
+                        <span class="headercat-title">new arrivals</span>
+                        <!--Хлебные крошки (делаются через сторонние программы)-->
+                        <span class="headercat-breadcrumbs"> home&nbsp;/ men<span class="color-pink">/ new
```
nav>ul>li>a

StasOskol replied:
<!--Второй блок под меню-->
                <nav class="bgcolorp">
                    <ul class="wrapper padding-title">
                        <li><a href="#" class="headercat-title">new arrivals</a></li>
                        <!--Хлебные крошки (делаются через сторонние программы)-->
                        <div class="headercat-navlink">
                            <li><a href="#" class="headercat-breadcrumbs">home</a></li>
                            <li><a href="#" class="headercat-breadcrumbs"> / men / </a></li>
                            <li><a href="#" class="headercat-breadcrumbs"><span class="color-pink">new arrivals</span></a></li>
                        </div>
                    </ul>
                </nav>

Я так понимаю вот так нужно было сделать?

VanillaWulf replied:
```
<nav>
<ul class="headercat-navlink">
                        <li><a href="#" class="headercat-breadcrumbs">home</a></li>
                        <li><a href="#" class="headercat-breadcrumbs"> / men / </a></li>
                        <li><a href="#" class="headercat-breadcrumbs"><span class="color-pink">new arrivals</span></a></li>
                    </ul>
</nav>
```

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+                                Completely visualize parallel core competencies rather than exceptional portals.
+                            </p>
+                            <span class="carousel-description-price">$561</span>
+                            <hr class="carousel-description-line">
```
это же можно марджином сделать и будет более удобно контролировать

StasOskol replied:
Имеется ввиду родительский блок описания? Ну я привязался к hr и задал padding, а внутри всё расположит по центру. Такая реализация плохая?

VanillaWulf replied:
отступ у предыдущего элемента

сетка - в css делается, так быстрее и проще ее править

VanillaWulf replied:
и сделайте черточку псевдоэлментом - это декор элемент

VanillaWulf commented on Product/Product.html:
```diff
@@ -0,0 +1,336 @@
+                        </div>
+                    </div>
+                </section>
+                <div class="wrapper wrapper-container margin-card">
```
я бы подумала, как эту обертку сделать в section - так будет аккуратнее по коду

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+    /*main*/
+
+    /*Карусель*/
+    .bgcolorprod{
```
пробел между название класса и скобкой - поправьте тут и далее

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+    /*Карусель*/
+    .bgcolorprod{
+        min-height: 777px;
+        background-color: #F7F7F7;
```
цвета обычно кладут в variables.scss

StasOskol replied:
Понял, это сделаю

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+            position: relative;            
+        }
+
+        &-btn{
```
сейчас мода на то, чтобы использовать & только для псевдоэлементов)
так как вы сделали тоже можно

StasOskol replied:
Ну за модой я не стремлюсь!)) Главное понимание, что делаешь

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+                display: block;
+                text-transform: uppercase;
+                font-weight: 300;
+                font-size: 14px;
```
шрифты подумайте как сделать через миксин

StasOskol replied:
Это сделаю

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+                line-height: 19px;
+                color: #FF6A6A;
+                background: white;
+                border: 1px solid #FF6A6A;
```
бордеры тоже можно через миксины делать

StasOskol replied:
Хорошо))

VanillaWulf commented on scss/style.scss:
```diff
@@ -816,4 +816,178 @@ $fillmenusvg:#E8E8E8;
+                display: flex;
+
+                &-svg{
+                    fill:#FF6A6A;
```
пробел

StasOskol replied:
К Вам вопрос: Вы случайно наткнулись на отсутствие пробела или через какую-то программу?

VanillaWulf replied:
случайно
можете попробовать beautify

StasOskol committed:
Ликвидация рамки при нажатии и наведении на кнопку

Строки для того, чтобы не выскакивала рамка
при нажатии и наведении на кнопку (ссылку)

StasOskol committed:
Создание новой папки css и  помещение файлов с раширением .css

Создание новой папки css и  помещение файлов с раширением .css

Удаление файлов с раширением .css в папке с препроцессором

StasOskol committed:
Исправление ошибки, что всё должно быть со строчных символов

Замена имени папок и файлов html,  на строчные символы

StasOskol committed:
Исправление ошибки с button, а так же подсключение стилей к станицам

Исправление button - надо указывать тип кнопок: type = "button"

StasOskol committed:
Исправление (замена div на section)

Не совсем понятно зачем это нужно, ещё попросили этот блок поменстить в main, но я не понимаю зачем (делать так не стал)

StasOskol committed:
nav>ul>li>a

nav>ul>li>a - задание изменить этот блок, не совсем понял зачем, но это сделал

StasOskol committed:
Разделение на подфайлы .scss (Применение Sass) , а также добавление библиотеки с шрифтами lato

StasOskol committed:
Досканальная работа с предпроцессором

1. Разделение на файлы препроцессора
2. Использование вложенности
3. Использование переменных
4. Разделение файла стилей на каждуую страницу
5. Использование миксинов
6. Подключение шрифтов с сочетанием миксинов
